app/src/main/java/com/sopt_seminar_4/View/LoginActivity.kt

```diff
@@ -0,0 +1,65 @@
+package com.sopt_seminar_4.View
+
+import android.content.ContentValues
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.sopt_seminar_4.Data.API.ApiLoginRegister
+import com.sopt_seminar_4.Data.DTO.LoginDTO.RequestLoginDTO
+import com.sopt_seminar_4.Data.Service.LoginService
+import com.sopt_seminar_4.databinding.ActivityLoginBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class LoginActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityLoginBinding
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        toRegisterPage()
+        loginRequest()
+    }
+
+    private fun toRegisterPage() {
+        binding.btnRegister.setOnClickListener {
+            startActivity(Intent(this@LoginActivity, RegisterActivity::class.java))
+        }
+    }
+
+    private fun loginRequest() {
+        binding.btnLogin.setOnClickListener {
+            ApiLoginRegister.create<LoginService>().also {
+
+                CoroutineScope(Dispatchers.Main).launch {
```

Comment:
코루틴 너무 좋습니다!

Reply:
```suggestion
                lifecycleScope.launch {
```

이렇게 해준다면 조금 더 안전하게 사용 할 수도 있을거 같아요

Reply:
어떤 차이인가요?!

Reply:
lifecyclescope를 해주게 되면 기타 동작은 동일하지만 액티비티의 생명주기를 따라가기 때문에 액티비티의 생명주기에 따라서 안전한 job cancel이 보장되기 때문에 더 안전합니다!
https://kotlinworld.com/198